sdcm/cluster_k8s/gke.py

```diff
@@ -157,15 +157,20 @@ def setup_gke_cluster(self, num_nodes: int) -> None:
 
     def add_gke_pool(self, name: str, num_nodes: int, instance_type: str) -> None:
         LOGGER.info("Create sct-loaders pool with %d node(s) in GKE cluster `%s'", num_nodes, self.name)
-        self.gcloud.run(f"container --project {self.gce_project} node-pools create {name}"
-                        f" --zone {self.gce_zone}"
-                        f" --cluster {self.gke_cluster_name}"
-                        f" --num-nodes {num_nodes}"
-                        f" --machine-type {instance_type}"
-                        f" --image-type UBUNTU"
-                        f" --node-taints role=sct-loaders:NoSchedule"
-                        f" --no-enable-autoupgrade"
-                        f" --no-enable-autorepair")
+        self.api_call_rate_limiter.put_requests_on_pause()
+        try:
+            self.gcloud.run(f"container --project {self.gce_project} node-pools create {name}"
+                            f" --zone {self.gce_zone}"
+                            f" --cluster {self.gke_cluster_name}"
+                            f" --num-nodes {num_nodes}"
+                            f" --machine-type {instance_type}"
+                            f" --image-type UBUNTU"
+                            f" --node-taints role=sct-loaders:NoSchedule"
+                            f" --no-enable-autoupgrade"
+                            f" --no-enable-autorepair")
+            self.kubectl('wait --timeout=15m --all --for=condition=Ready node')
+        finally:
+            self.api_call_rate_limiter.release_requests_pause()
```

Comment:
What do you think about using RLock() instead of event and use it as context manager:

```python
with self.api_call_rate_limiter.pause:
    ...
```
here, and 

```python
...
if not self.pause.locked() and self._lock.get_value() == 0:
...
```
in `def run(self):` later

?

Reply:
Functionally RLock is worse suited for this particular purpose where we need to have one place for commanding to put any next request on pause.
But context manager idea looks good, redone in cm manner.